src/components/message-composer.tsx

```diff
@@ -1,4 +1,31 @@
-function MessageComposer() {
+import { useState } from "preact/hooks"
+
+import { supabase } from "../supabaseClient"
+
+type MessageComposerType = {
+  channelId: number
+  charmerId: string
+}
+
+function MessageComposer({ channelId, charmerId }: MessageComposerType) {
+  const [content, setContent] = useState<string>()
+
+  const handleChange = (event: any) => {
+    setContent(event.currentTarget.value)
+  }
+
+  const handleKeyPress = async (event: { key: string }) => {
+    if (event.key === "Enter") {
+      const { error } = await supabase
+        .from("messages")
+        .insert({ channel_id: channelId, charmer_id: charmerId, content })
+
+      if (error) console.error(error)
+
+      setContent("")
```

Comment:
if we have an error the message will be lost, right? maybe we need an else statement here 🤔

Reply:
Totally! We should add a toast or some kind of general error handler, app-wide, what do you think?